### ISSUE TITLE
Page doesn't allow for Retemplating 

### ISSUE THREAD

**Issue by DotNetRussell** [microsoft/microsoft-ui-xaml#1075]
**Describe the bug**
I attempt to subclass page and retemplate it using a templated control. However, it doesn't currently work. The consumer's content overrides the templated controls style. 

**Steps to reproduce the bug**
1. Open a new blank UWP app
2. Right click on the csproj and add a new templated control named `TestPage.xaml`
3. Open `TestPage.xaml.cs` and sublcass Page
4. Open `Themes/Generic.xaml` 
5. In the newly generated style of `TestPage`, remove the default border control and replace it with the following

```
  <TextBlock Text="Templated Content" />                        
  <ContentPresenter />
```

6. Open `MainPage.xaml` and change the type of the page to `TestPage` 
7. Open `MainPage.xaml.cs` and subclass `TestPage`
8. In the body of  `MainPage.xaml`, add a `TextBlock` that has the text `Consumer Content`
9. Run the application
NOTE: Only `Consumer Content` is visible in the application.

**Expected behavior**
Both the consumer content and the templated content should be visible 

**Version Info**
Targeting all versions 

Only tested on Desktop

**Additional context**
Related SO Post https://stackoverflow.com/questions/57064473/creating-a-subclass-of-page-in-uwp

Related Github Repo https://github.com/DotNetRussell/UWP_Page_Inheritance_Bug/tree/master/DemoBasePageBug

Twitter thread https://twitter.com/DotNetRussell/status/1151530698098061312

**Comment by jevansaks** [comment]
I believe that is because Page is a UserControl so the Content property is the only thing that's visible. So this is currently by design.

What is the scenario for this?

**Comment by DotNetRussell** [comment]
> I believe that is because Page is a UserControl so the Content property is the only thing that's visible. So this is currently by design.
> 
> What is the scenario for this?

The scenario is creating a pre designed page that has default chrome that the consumer would get when they used this page. For example maybe I want the page to have an application notification control, or a status bar or a context ribbon. I don't want my consumer to have to implement that in every page. I would rather implement it once, then have the consumer use my page over and over.

**Comment by mrlacey** [comment]
> The scenario is creating a pre designed page that has default chrome that the consumer would get when they used this page. For example maybe I want the page to have an application notification control, or a status bar or a context ribbon. I don't want my consumer to have to implement that in every page. I would rather implement it once, then have the consumer use my page over and over.

Possible work-around (depending on your exact requirements):
- You could retemplate `frame` to have the global elements in it, rather than having copies of them in each page.
- Create a page that has the desired global elements in it and host a frame within that that loads all the other pages.

Having a single version of the global elements, rather than copies of them on each page, has made things simpler for me in past projects.

**Comment by DotNetRussell** [comment]
@mrlacey I agree that both of those should be work arounds. Unfortunately, this is an existing enterprise application that has existing design choices that prevent this.

**Comment by grork** [comment]
Based on my prior experience with this, and attempting to solve a similar problem across 3 separate apps: Create a custom control, with it's own template, and use that as a wrapper around the content of the page (`Page->(New Control)->Whatever you'd put in your page normally`). This enables 3 use cases:
1. Your primary use case of consistency for page-level framing
2. Because your 'wrapper' control has it's own template (not user control, template), it allows page-level customizations of visuals, and maintaining behaviour. I've used this a number of times to solve unique product requirements.
3. When some has a Good Reason™, they won't want the page level controls. In this model, they just leave the wrapper control out.

I have tried the frame level model in the past, and the "if's, buts, and maybes" that come down from product & biz requirements result in a spaghetti code in that frame to control all the considerations in 2.
